cassandra/cqltypes.py

```diff
@@ -291,7 +291,11 @@ def cass_parameterized_type_with(cls, subtypes, full=False):
             cname = apache_cassandra_type_prefix + cname
         if not subtypes:
             return cname
-        sublist = ', '.join(styp.cass_parameterized_type(full=full) for styp in subtypes)
+        fieldnames = getattr(cls, 'fieldnames', None)
+        if fieldnames and any(fieldnames):
```

Comment:
I'm a little cold on this change. I thought I had reasoned that fieldnames would only be set on this path for DCT, but it's a little hazy now. Instead of this, I can just specialize the method for DCT. I think that should be more clear.